samples/subsys/ipc/rpmsg_demo/src/rsc_table.cpp

```diff
@@ -0,0 +1,63 @@
+#include "rsc_table.hpp"
+#include <openamp/virtio.h>
+#include <stddef.h>
+#include <device.h>
+
+#define RSC_VDEV_FEATURE_NS (1) /* Support name service announcement */
+#define RSC_VDEV_NUM_VRINGS (2)
+
+#define VDEV0_VRING_BASE (DT_REG_ADDR(DT_NODELABEL(vdev0vring0)))
+#define VDEV1_VRING_BASE (DT_REG_ADDR(DT_NODELABEL(vdev0vring1)))
+
+#define VRING_ALIGN (0x1000U)
+#define RL_BUFFER_COUNT (256U)
+#define RESOURCE_TABLE_OFFSET (0xFF000)
+
+Z_GENERIC_SECTION(.resource_table)
+remote_resource_table resources = {
+    .version = 1,
+    .num = RSC_TABLE_NUM_ENTRY,
+    .offset = { offsetof(remote_resource_table, user_vdev), },
+
+    /* SRTM virtio device entry */
+    .user_vdev = {
+        .type = RSC_VDEV,
+        .id = VIRTIO_ID_RPMSG,
+        .notifyid = 0,
+        .dfeatures = RSC_VDEV_FEATURE_NS,
+        .gfeatures = 0,
+        .config_len = 0,
+        .status = 0,
+        .num_of_vrings = RSC_VDEV_NUM_VRINGS,
+    },
+
+    /* Vring rsc entry - part of vdev rsc entry */
+    .user_vring0 = {
+        .da = VDEV0_VRING_BASE,
+        .align = VRING_ALIGN,
+        .num = RL_BUFFER_COUNT,
+        .notifyid = 0,
+    },
+
+    .user_vring1 = {
+        .da = VDEV1_VRING_BASE,
+        .align = VRING_ALIGN,
+        .num = RL_BUFFER_COUNT,
+        .notifyid = 1
+    },
+};
+
```

Comment:
A resource table is defined here: https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/open-amp/resource_table.c
could it be reused?

Reply:
I think it can, however some differences:

- `vring` alignment is set to `0x1000` instead of `0x10/16`
- `vring` addresses are taken from the devicetree instead of `-1`
- The resource table in the NXP mcuxpresso example is copied to 2 location for some reason ([copyResourceTable](https://github.com/zephyrproject-rtos/zephyr/pull/42967/files#diff-6e4bf27b11af212ff9edbfb2bd79748b0eccf3ead89fccaec5db747db0a7374cR50))

Reply:
> I think it can, however some differences:
> 
> * `vring` alignment is set to `0x1000` instead of `0x10/16`
> * `vring` addresses are taken from the devicetree instead of `-1`

right these parameters should be customized depending on Hardware

> * The resource table in the NXP mcuxpresso example is copied to 2 location for some reason ([copyResourceTable](https://github.com/zephyrproject-rtos/zephyr/pull/42967/files#diff-6e4bf27b11af212ff9edbfb2bd79748b0eccf3ead89fccaec5db747db0a7374cR50))

The reason for this seems rather obscure to me. But yes, it seems specific to IMX hardware.